### PR TITLE
cleanup(rules)!: refactor kernel module rule

### PR DESCRIPTION
**Pull request by incertum**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area rules

> /area registry

> /area build

> /area documentation

**What this PR does / why we need it**:

Rule "Linux Kernel Module Injection Detected" is limited in scope.
A major refactor of Falco now exposes each syscall Falco's libs supports (including `init_module`) to the end user :)
Official support starts with Falco 0.35.0.

**Which issue(s) this PR fixes**:

https://github.com/falcosecurity/falco/issues/2443#issuecomment-1471289452

Fixes #

**Special notes for your reviewer**:

CC @loresuso @LucaGuerra @jasondellaluce 

```
cleanup(rules)!: refactor kernel module rule
```

### CODE REVIEW

File: rules/falco_rules.yaml [hunk 1]
```diff
@@ -3150,10 +3153,15 @@
...
 
+# init_module syscall is available since Falco 0.35.0
+# rule coverage now extends to modprobe usage
 - rule: Linux Kernel Module Injection Detected
   desc: Detect kernel module was injected (from container).
-  condition: spawned_process and container and proc.name=insmod and not proc.args in (white_listed_modules) and thread.cap_effective icontains sys_module
-  output: Linux Kernel Module injection using insmod detected (user=%user.name user_loginuid=%user.loginuid parent_process=%proc.pname module=%proc.args %container.info image=%container.image.repository:%container.image.tag)
+  condition: ((spawned_process and proc.name=insmod) or init_module)
+    and container
+    and not proc.args in (white_listed_modules)
```
Comment (loresuso) [inline rules/falco_rules.yaml:3162]: Nice, thanks for noticing!
Btw, I am thinking that maybe it's time to change this rule and make it even better.
I would like to drop the part of the condition relying on `insmod`, and use instead the related syscalls, which is of course `init_module` but also `finit_module` (which is what my `insmod` is using). wdyt? :)

Comment (incertum) [reply rules/falco_rules.yaml]: Thank you @loresuso this is great! Done :)

Comment (loresuso) [reply rules/falco_rules.yaml]: Nice! One last thing is either throw away the list `white_listed_modules` cause now it should be unused.
I'd add an `allowed_container_images_loading_kernel_module` list and add it to the condition with `and not` and it's perfect for me!

File: rules/falco_rules.yaml [hunk 2]
```diff
@@ -3147,13 +3150,14 @@
   enabled: false
   tags: [host, container, network, mitre_command_and_control, TA0011]
 
-- list: white_listed_modules
```
Comment (Kaizhe) [inline rules/falco_rules.yaml:3150]: who do we remove the whitelist ?

Comment (loresuso) [reply rules/falco_rules.yaml:3150]: For me now given that the rule is changing, it makes sense having a whitelist on *container images* that can load modules

Comment (Kaizhe) [reply rules/falco_rules.yaml:3150]: did you add it in the condition

Comment (loresuso) [reply rules/falco_rules.yaml:3150]: no, that's what I was asking (see above), let's wait for Melissa :)

Comment (Kaizhe) [reply rules/falco_rules.yaml:3150]: with the whitelisted images (e.g. falco), I think we're good to go.

Comment (incertum) [reply rules/falco_rules.yaml:3150]: Added, this is great feedback, ty :)

**Review by loresuso — APPROVED** [review]
Nice! /approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Comment by jasondellaluce** [comment]
Note for us: renaming the macro can represent a breaking change. Let's keep this in mind for the next falco_rules release.

**Comment by Kaizhe** [comment]
> Note for us: renaming the macro can represent a breaking change. Let's keep this in mind for the next falco_rules release.

Maybe let's revert the macro, we should probably give a heads up in the release notes (saying this macro will be deprecated in release 3.8 or 9)

**Pull request merged**